### PR TITLE
feat: implement the adena wallet provider

### PR DESCRIPTION
**Pull request by jinoosss**
### What type of PR is this?

- feature

### What this PR does:
- Define the wallet provider that will be managed in your SDK instance, and implement the adena wallet provider.

- Change the adena methods to provider.

### CODE REVIEW

**Comment by dongwon8247** [comment]
@RezaRahemtola Can you give it a review?

File: packages/sdk/package-lock.json
Comment (RezaRahemtola) [inline packages/sdk/package-lock.json:1]: is this needed ? there's already a `yarn.lock` at the root

File: packages/sdk/src/core/types/wallet.types.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  TRANSACTION_FAILED = 'TRANSACTION_FAILED',
+  SIGN_FAILED = 'SIGN_FAILED',
+  ALREADY_CONNECTED = 'ALREADY_CONNECTED',
+  NETWORK_TIMEOUT = 'NETWORK_TIMEOUT',
+  REDUNDANT_CHANGE_REQUEST = 'REDUNDANT_CHANGE_REQUEST',
+  NETWORK_ALREADY_EXISTS = 'NETWORK_ALREADY_EXISTS',
+  UNADDED_NETWORK = 'UNADDED_NETWORK',
+  UNSUPPORTED_TYPE = 'UNSUPPORTED_TYPE',
+  UNEXPECTED_ERROR = 'UNEXPECTED_ERROR',
+}
```
Comment (RezaRahemtola) [inline packages/sdk/src/core/types/wallet.types.ts:41]: should we unify those types in the future in a common package used by the main Adena repo and this SDK ?

**Review by RezaRahemtola — COMMENTED** [review]
Code looks good to me otherwise, it's going in the right direction!
I'll test this new SDK architecture locally tomorrow with one of my Gno projects to give more feedbacks regarding usage

File: packages/sdk/package-lock.json
Comment (jinoosss) [reply packages/sdk/package-lock.json:1]: This file is clearly unnecessary.

File: packages/sdk/src/core/types/wallet.types.ts (hunk 1 above)
Comment (jinoosss) [reply packages/sdk/src/core/types/wallet.types.ts:41]: Yes, I would like to unify the types and responses.
If the SDK and wallet responses were separate, it wouldn't be a problem if the provider did the mapping for me, but I'd have to manage them separately and always consider what I'm missing.

**Review by dongwon8247 — APPROVED** [review]

**Comment by jinoosss** [comment]
@RezaRahemtola 
Thanks for the review !

We have a PR that implements the features of the SDK.
We would be grateful if you could review it in the environment of this PR(https://github.com/onbloc/adena-wallet-sdk/pull/5).

**Pull request merged**